webapp/src/components/PollsPage/PollsPage.tsx

```diff
@@ -1,136 +1,125 @@
+  expiredRows: number = 10
+  shouldFetchActivePolls: boolean = false
+  shouldFetchExpiredPolls: boolean = false
 
-export default class PollsPage extends React.PureComponent<PollsPageProps> {
   componentWillMount() {
-    this.props.onFetchPolls()
+    this.fetchActivePolls()
+    this.fetchExpiredPolls()
   }
 
-  render() {
-    const { polls, wallet, isLoading } = this.props
+  componentWillReceiveProps({ location: { search } }: Props) {
+    const query = queryString.parse(search)
+    const { activePage, expiredPage } = this.getPagination()
+    const newActivePage = +query.active
+    const newExpiredPage = +query.expired
+    if (!isNaN(newActivePage) && newActivePage !== activePage) {
+      this.shouldFetchActivePolls = true
+    }
+    if (!isNaN(newExpiredPage) && newExpiredPage !== expiredPage) {
+      this.shouldFetchExpiredPolls = true
+    }
+  }
 
-    const districtPolls = Object.values(polls)
-      .filter(poll => isDistrictToken(poll.token))
-      .sort(sortByContributions(wallet))
-    const dclPolls = Object.values(polls).filter(
-      poll => !isDistrictToken(poll.token)
-    )
+  componentDidUpdate() {
+    if (this.shouldFetchActivePolls) {
+      this.fetchActivePolls()
+      this.shouldFetchActivePolls = false
+    }
+    if (this.shouldFetchExpiredPolls) {
+      this.fetchExpiredPolls()
+      this.shouldFetchExpiredPolls = false
+    }
+  }
 
+  fetchActivePolls() {
+    const { onFetchPolls } = this.props
+    const { activePage } = this.getPagination()
+    onFetchPolls({
+      limit: this.activeRows,
+      offset: (activePage - 1) * this.activeRows,
+      active: true,
+      expired: false
+    })
+  }
+
+  fetchExpiredPolls() {
+    const { onFetchPolls } = this.props
+    const { expiredPage } = this.getPagination()
+    onFetchPolls({
+      limit: this.expiredRows,
+      offset: (expiredPage - 1) * this.expiredRows,
+      active: false,
+      expired: true
+    })
+  }
+
+  getQueryParams() {
+    const {
+      location: { search }
+    } = this.props
+    return queryString.parse(search)
+  }
+
+  getPagination() {
+    const queryParams = this.getQueryParams()
+    return {
+      activePage: +queryParams.active || 1,
+      expiredPage: +queryParams.expired || 1
+    }
+  }
+
+  handleActivePageChange = (page: number) => {
+    const { onPageChange } = this.props
+    const { expiredPage } = this.getPagination()
+    onPageChange(page, expiredPage)
+  }
+
+  handleExpiredPageChange = (page: number) => {
+    const { onPageChange } = this.props
+    const { activePage } = this.getPagination()
+    onPageChange(activePage, page)
+  }
+
+  render() {
+    const {
+      activePolls,
+      expiredPolls,
+      isLoading,
+      totalActive,
+      totalExpired
+    } = this.props
+    const { activePage, expiredPage } = this.getPagination()
     return (
       <div className="PollsPage">
-        {isLoading ? (
+        {isLoading && activePolls.length === 0 && expiredPolls.length === 0 ? (
```

Comment:
I know it's a bit far fetched but wouldn't this be loading forever if we clean the polls?

Reply:
mm it's an `&&` so it would be loading only while the request is pending. This is to avoid showing the spinner when you change the page

Reply:
my bad!